src/Modals/ModalNewForm/ModalNewForm.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react';
+import "./ModalNewForm.css";
+import { connect } from 'react-redux';
+import Actions from "../../Actions/actions";
+
+
+class PostForm extends Component {
+    constructor() {
+        super();
+        this.state = {
+            'name': '',
+            'address_street': '',
+            'address_apt': '',
+            'address_city': '',
+            'note': '',
+            'date': '',
+        }
+    }
+
+    isSubmitButtonEnable(){
+        return ((this.state.name.length > 0) && (this.state.address_street.length > 0) && (this.state.address_city.length > 0))
+    }
+
+    getDate(){
+        let date = new Date();
+        let month = date.getUTCMonth() + 1;
+        let day = date.getUTCDate();
+        let year = date.getUTCFullYear();
+        return month + "/" + day + "/" +  year;
+    }
+
+    onChangeInput(type, answer){
+        this.setState({
+            [type]: answer,
+        })
+
+    }
+
+    clearState(){
+        this.setState({
+            'name': '',
+            'address_street': '',
+            'address_apt': '',
+            'address_city': '',
+            'note': '',
+            'date': '',
+        })
+    }
+
+    handleSubmit(){
+        const address = `${this.state.address_street} ${this.state.address_apt}, ${this.state.address_city}`
+        const postInfo = {
+           'date': this.getDate(),
+           'name': this.state.name,
+           'address': address,
+           'note': this.state.note
+       }
+       if(this.isSubmitButtonEnable()){
+        this.props.addNewPost(postInfo)
+        this.props.toggleModal('newForm', false)
```

Comment:
This is something that could be done through sagas! You're doing 2 actions when submit is clicked right? But toggling the modal isn't the **main** action, it's a side effect. So it'd be cleaner to have a saga watch for whenever a new post is added, close the modal.